### ISSUE TITLE
[Feature Request] Support conditional sampling with logical operators (e.g., >, <, !=) using Constrained Decoding (Logit Masking)

### ISSUE THREAD

**Issue by unnir** [kathrinse/be_great#62]
Currently, GReaT supports conditioning the generation process on specific, exact values (e.g., `starting_prompts="Age is 30"`). However, users often need to generate synthetic data that satisfies specific ranges or inequalities, such as "Users older than 30" or "Income between 50k and 100k".

We want to extend the `.sample()` method to accept a dictionary of conditions with mathematical/logical operators.

We think this API will be great:

```python
# Example Usage
synthetic_data = model.sample(
    n_samples=100, 
    conditions={
        "Age": ">= 30",
        "Income": "< 50000",
        "City": "!= 'New York'"
    }
)
```

Suggested Implementation Strategy

One idea is to utilize Hugging Face's LogitsProcessor or LogitsWarper to modify the generation probabilities on the fly.

See here: https://huggingface.co/docs/transformers/en/internal/generation_utils

Issue: Since numbers are often tokenized into multiple sub-tokens (e.g., 100 might be 1, 0, 0), the processor needs to handle partial number construction. A Prefix Trie approach or integration with a library like guidance concepts (adapted for our specific usage) might be necessary.

## Requirements 

 - Zero Waste: The model should not generate a full row that violates the condition, only to discard it later. The constraint must be applied during generation.
- Operators: Support >. <, >=, <=, ==, !=  
- Token Handling: Must correctly handle multi-token numbers (e.g., ensuring that if > 100 is required, the model doesn't stop after generating 1).
- Integration: Must pass as a logits_processor list to the underlying self.model.generate() call in great.py.

Help Wanted!

**Comment by Sakthivel0408** [comment]
I'll take this on and start working on it.
